App/src/Service/ExpenseBalancer.php

```diff
@@ -7,13 +7,14 @@
 use App\Entity\User;
 use App\Entity\UserBalance;
 use App\Repository\UserBalanceRepository;
+use Doctrine\Common\Collections\Collection;
 use Doctrine\ORM\EntityManagerInterface;
 
 class ExpenseBalancer
 {
     public function __construct(
         private readonly EntityManagerInterface $entityManager,
-        private readonly UserBalanceRepository $userBalanceRepository
+        private readonly UserBalanceRepository $userBalanceRepository,
     ) {
     }
```

Comment:
When all your attributes are readonly you can make the entire class readonly. 😉 Except those which extends a non-readonly class